proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1033 @@ If WiFi credentials are automatically generated by the IVI system, the password
+<!--
+@startuml ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multiapps
+!include ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multiapps.puml
+@enduml
+-->
+![Vehicle access point multiple apps](../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multiapps.png)
 
-Here is an example of the WiFi configuration APIs to set WiFi configuration in Android proxy:
+### Multiple device connection
 
-```java
-/** API to make mobile device as WiFi access point*/
-MultiplexTransportConfig.setCanHostWiFiNetwork(Boolean);
+#### Vehicle is a network host
 
-/** API to automatically connect WiFi access point created by IVI*/
-MultiplexTransportConfig.setCanAutoJoinWiFiNetwork(Boolean);
-```
+For each connected device, SDL Core will select a single feature enabled application.
+
+Once network parameters are established with the first device, `OnNetworkConfiguration` will be sent to this application and HMI. After receiving this notification, HMI should send `OnSystemCapabilityUpdated` in case capabilities have changed. The next application's capabilities will be compared to updated HMI capabilities.
+
+This mechanism will allow the HMI to decide its own behavior based on all of the following incoming connections.
 
-3. The proxy should set `DeviceInfo.wifiConfigurationInfo` with appropriate values.
+<!--
+@startuml ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multidev
+!include ../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multidev.puml
+@enduml
+-->
+![Vehicle access point multiple devices](../assets/proposals/0245-sharing-wifi-ssid-and-password/vehicle_access_point_multidev.png)
 
-Here is an example of the WiFi configuration APIs usage to set `DeviceInfo.wifiConfigurationInfo` in Android proxy:
 
-* If `MultiplexTransportConfig.setCanHostWiFiNetwork( true )` and mobile device __DOES__ support WiFi transport APIs provided by platform SDK then `DeviceInfo.wifiConfigurationInfo.canHostWiFiNetwork = true `
-* If `MultiplexTransportConfig.setCanAutoJoinWiFiNetwork( true )` and mobile device __DOES__ support WiFi transport APIs provided by platform SDK then `DeviceInfo.wifiConfigurationInfo.canAutoJoinWiFiNetwork = true`
+#### Mobile device is a network host
 
-5. The proxy should be able to check the mobile device OS version. In case platform API support to connect WiFi is not available in a mobile device, the proxy should be able to set `DeviceInfo.wifiConfigurationInfo` values to `false` at run time.
+A precondition for further discussion is that the first device is connected, it hosts the network and HMI successfully connected to its hotspot. SDL Core already received `OnSystemCapabilityUpdate` with updated networking capabilities.

```

Comment:
All references to `OnSystemCapabilityUpdate` should be `OnSystemCapabilityUpdated`

Reply:
Fixed with 8b3217b